packages/shared/src/features/profile/components/ProfileSquadsWidget.tsx

```diff
@@ -0,0 +1,205 @@
+import React from 'react';
+import type { Squad } from '../../../graphql/sources';
+import { useAuthContext } from '../../../contexts/AuthContext';
+import { labels, largeNumberFormat } from '../../../lib';
+import {
+  Button,
+  ButtonVariant,
+  ButtonSize,
+  ButtonIconPosition,
+} from '../../../components/buttons/Button';
+import { StraightArrowIcon } from '../../../components/icons';
+import { IconSize } from '../../../components/Icon';
+import { webappUrl } from '../../../lib/constants';
+import { anchorDefaultRel } from '../../../lib/strings';
+
+import { LazyImage } from '../../../components/LazyImage';
+import {
+  Typography,
+  TypographyType,
+  TypographyColor,
+  TypographyTag,
+} from '../../../components/typography/Typography';
+import { useViewSize, ViewSize, useToastNotification } from '../../../hooks';
+import { SquadActionButton } from '../../../components/squads/SquadActionButton';
+import { Origin } from '../../../lib/log';
+import { useToggle } from '../../../hooks/useToggle';
+import { ActivityContainer } from '../../../components/profile/ActivitySection';
+import {
+  useSources,
+  getFlatteredSources,
+} from '../../../hooks/source/useSources';
+
+interface ProfileSquadsWidgetProps {
+  squads: Array<Squad>;
+  userId: string;
+}
+
+interface SquadListItemProps {
+  squad: Squad;
+}
+
+const MOBILE_MAX_SQUADS = 3;
+const DESKTOP_MAX_SQUADS = 5;
+
+const useProfileSquadsWidget = (props: ProfileSquadsWidgetProps) => {
```

Comment:
This _local hook_ has basically the whole logic, and send back all that is needed by the component